src/generic/stage1/utf8_validator.h

```diff
@@ -0,0 +1,26 @@
+namespace stage1 {
+/**
+ * Validates that the string is actual UTF-8.
+ */
+template<class checker>
+bool utf8_validate(const uint8_t * input, size_t length) {
+    checker c{};
+    buf_block_reader<64> reader(input, length);
```

Comment:
We can and should check perf later: is it possible the utf-8 validator will go faster with 128 byte blocks, given that stage 1 did? Maybe even more given the utf-8 algorithm is smaller ...

Reply:
Right. This is an internal issue and could be tweaked later.